solver/src/driver.rs

```diff
@@ -197,29 +163,94 @@ impl Driver {
                 continue;
             }
 
-            let trades = settlement.trades().to_vec();
+            let trades = settlement.settlement.trades().to_vec();
             match settlement_submission::submit(
                 &self.settlement_contract,
                 self.gas_price_estimator.as_ref(),
                 self.target_confirm_time,
                 GAS_PRICE_CAP,
-                settlement,
+                settlement.settlement,
             )
             .await
             {
                 Ok(_) => {
-                    // TODO: order validity checks
-                    // Decide what is handled by orderbook service and what by us.
-                    // We likely want to at least mark orders we know we have settled so that we don't
-                    // attempt to settle them again when they are still in the orderbook.
```

Comment:
Removed comment because it's not relevant anymore or will only be relevant when do speculative settlements (relying on previous going through) in the future.